contrib/react-native-debugging.md

```diff
@@ -0,0 +1,110 @@
+$ cd ReactTestApp
+$ npm install
+$ npx install-local
+$ npx pod-install
+
+# open the project in xcode
+$ open ios/ReactTestApp.xcworkspace
+```
+
+You should now be able to move onto
+- [Debugging Javascript](#debugging-javascript) 
+- [Debugging C++](#debugging-c++) 
+
+## Setup Custom React Native Project
+
+### Prepare Javascript Debugging
+In order to view Javascript in Safari, it is required to make a small change to your projects AppDelegate.m.
+
+```objective-c
+- (NSURL *)sourceURLForBridge:(RCTBridge *)bridge
+{
+#if DEBUG
+  // This is the original return from React Native
+  //return [[RCTBundleURLProvider sharedSettings] jsBundleURLForBundleRoot:@"index" fallbackResource:nil];
+
+  // This return allows for debugging the JavaScript in Safari
+  return [NSURL URLWithString:[[[[RCTBundleURLProvider sharedSettings] jsBundleURLForBundleRoot:@"index" fallbackResource:nil] absoluteString] stringByAppendingString:@"&inlineSourceMap=true" ]];
+#else
+  return [[NSBundle mainBundle] URLForResource:@"main" withExtension:@"jsbundle"];
+#endif
+}
+```
+
+Once this is done, it is possible to [debug javascript](#debugging-javascript) in Safari.
+
+### Prepare C++ Debugging
+To prepare for c++ debugging, right click on your project in xcode and select "Add Files to 'projectname'...".  In the dialog, find the `src` folder in your `realm-js` project. Make sure that all targets are deselected, "Copy items if needed" is deselected and "Create folder refernces" is selected.
+
+![Add files dialog](./assets/addFilesDialog.png)
+
+The project is now prepared to [debug c++](#dubugging-c++).
+
+### Reference realm-js Project
+
+The easiest way to get the checked out realm-js project into the React Native project is to invoke:
+```
+$ node ./scripts/watch-from-dependency.js -p <Path To ReactNative App>
+```
+
+This will copy the project into the node_modules folder and monitor for changes.
+
+###
+## Debugging Javascript
+
+The project should now be prepared to debug javascript.  Safari will be needed to do this. The first step is to activate the Develop menu item:
+
+![Safari Prefrences](./assets/safariPreferences.png)
+
+Now start running the app in the iOS simulator, either from Xcode or the command line.  When the app is started, click the Develop menu item in safari, highlight the running simulator name and click JSContext.  Selecting "Automatically Show Web Inspector for JSContexts" will make sure the inspector is shown after reloads to the app.
+
+![Select JSContext](./assets/selectJscontext.png)
+
+In the inspector select the Sources tab.  The index.bundle can now be expanded to view the source directories.  Find the desired js file and set some breakpoints to debug.
+
+![Safari Inspector](./assets/safariInspector.png)
+
+
+<b>NOTE</b>: if in the process of this guide, the javascript is not visible, it may be necessary to download and use [Safari Technology Preview](https://developer.apple.com/safari/technology-preview/) instead
```

Comment:
```suggestion
<b>NOTE</b>: if, in the process of this guide, the JavaScript source files are not visible, it may be necessary to download and use [Safari Technology Preview](https://developer.apple.com/safari/technology-preview/) instead
```